### PR TITLE
lib/osutil: Don't attempt to reduce our niceness level (fixes #4681)

### LINKED ISSUE #4681: "Failed to lower process priority: set niceness: permission denied" when already niced
```
$ nice -15 ./syncthing -home test
...
[4DVPI] 13:45:31 WARNING: Failed to lower process priority: set niceness: permission denied
```

### PR DESCRIPTION
**Pull request by calmh**
### Purpose

Avoid complaining to the user about permission denied when this happens.

### CODE REVIEW

**Comment by calmh** [comment]
@st-review don't

**Comment by st-review** [comment]
@calmh: Preventing merge for the time being. Push a new revision to reset!

File: lib/osutil/lowprio_linux.go [hunk 1]
```diff
@@ -42,23 +42,33 @@ func ioprioSet(class ioprioClass, value int) error {
...
 // I/O priority depending on the platform and OS.
 func SetLowPriority() error {
+	// Process zero is "self", niceness value 9 is something between 0
+	// (default) and 19 (worst priority).
+	const (
+		pidSelf       = 0
+		wantNiceLevel = 9
+	)
+
+	if cur, _ := syscall.Getpriority(syscall.PRIO_PROCESS, pidSelf); cur >= wantNiceLevel {
```
Comment (AudriusButkevicius) [inline lib/osutil/lowprio_linux.go]: Why are we not handling the error?

Comment (calmh) [reply lib/osutil/lowprio_linux.go]: I'm being lazy and assuming that if there is an error I'm not to get a positive integer back

**Comment by calmh** [comment]
Gnnh

```
   C library/kernel differences
       Within  the  kernel,  nice  values  are actually represented using the range 40..1 (since negative numbers are
       error codes) and these are the values employed by the setpriority() and getpriority() system calls.  The glibc
       wrapper  functions for these system calls handle the translations between the user-land and kernel representa‐
       tions of the nice value according to the formula unice = 20 - knice.  (Thus, the kernel's 40..1  range  corre‐
       sponds to the range -20..19 as seen by user space.)
```

File: lib/osutil/lowprio_linux.go (hunk 1 above)
Comment (imsodin) [reply lib/osutil/lowprio_linux.go]: Wouldn't it make sense to abort if the niceness is anything but the default value? Line in if the user sets niceness manually, it's nicer (haha) to not mess with it at all.

**Comment by calmh** [comment]
I think this works properly now

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
:ok_hand: Merged as 3af4164c8bd89c4f9d881c664e9bcb4b4586e875. Thanks, @calmh!

**Pull request closed without merging**